ebitenutil/shapes.go

```diff
@@ -59,3 +60,25 @@ func DrawRect(dst *ebiten.Image, x, y, width, height float64, clr color.Color) {
 	// Linear filtering would make edges blurred.
 	dst.DrawImage(emptyImage.SubImage(image.Rect(1, 1, 2, 2)).(*ebiten.Image), op)
 }
+
+// DrawCircle draws a circle on given destination dst.
+//
+// DrawCircle is intended to be used mainly for debugging or prototyping puropose.
+func DrawCircle(dst *ebiten.Image, cx, cy, r float32, clr color.Color) {
+	var path vector.Path
+	rd, g, b, a := clr.RGBA()
+
+	path.Arc(cx, cy, r, 0, 2*math.Pi, vector.Clockwise)
+	op := &ebiten.DrawTrianglesOptions{
+		FillRule: ebiten.FillAll,
+	}
+
+	verticles, indices := path.AppendVerticesAndIndicesForFilling(nil, nil)
+	for i := range verticles {
+		verticles[i].ColorR = float32(rd) / 0xffff
+		verticles[i].ColorG = float32(g) / 0xffff
+		verticles[i].ColorB = float32(b) / 0xffff
+		verticles[i].ColorA = float32(a) / 0xffff
```

Comment:
Shouldn't vertices' SrcX/SrcY be set?

Reply:
I don't think it has to
I think `vector.Path.AppendVerticesAndIndicesForFilling` does not do it by default it's still drawing
Here SrcX, SrcY will be fixed 0 in all interpolated points of the triangles, so it will map to 0,0 from the white filled image so the color will be white
I don't think it matters since we're drawing right after that (l.83) and controlling the input image

Reply:
> Also it would probably not make sense if it was mapped to an asset image, if we imagine what kind of "triangles" compose the final shape

As the source image is the emtpySubImage whose bound is (1,1)-(2,2), we should set SrcX/SrcY to do so. I think setting all the SrcX/SrcY to (1, 1) would work at least.

Reply:
> As the source image is the emtpySubImage whose bound is (1,1)-(2,2)

Oh, you're right I didn't realize the source image was a sub image, makes sense!

Reply:
Yup, thanks for your explanation @hajimehoshi didn't know that :D, I will make fix. But I wonder what really difference it makes because it works either way ;)

Reply:
Yeah, (0, 0) would work in 99% cases but this is unsafe unfortunately. I used a 1x1 sub image inside of a 3x3 white image to prevent bleeding edges. Bleeding edges are a common GPU issue that an outside pixel from the image region is picked. This is inevitable for any GPUs. By using a subimage whose bounds are (1,1)-(2,2), picking outside pixels should not be problematic.

Reply:
Done ;)